OBSDOCS-1859: Loki Query performance troubleshooting steps (#99687)
openshift/openshift-docs: modules/best-practices-for-loki-query-performance.adoc

:_newdoc-version: 2.18.4
:_template-generated: 2025-09-25
:_mod-docs-content-type: CONCEPT

[id="best-practices-for-loki-query-performance_{context}"]
= Best practices for Loki query performance

> The following are the best practices for Loki query performance:

Reviewer: ```suggestion
The following steps contribute to improve Loki query performance:
```

> * Ensure that you are running the latest version of the operator.

Reviewer: ```suggestion
* Ensure that you are running the latest version of the loki-operator.
```

* Ensure that you have migrated LokiStack schema to the `v13` version.

> * Ensure that you use reliable and fast object storage. Loki is a database capable of doing a tremendous amount of work on object storage.

Reviewer: ```suggestion
* Ensure that you use reliable and fast object storage. Loki is a database capable of doing a tremendous amount of work on object storage. If you're not using an object store solution from a cloud provider, we highly recommend the use of SSD's as the backend of your object storage, due to the high parallelization that Loki is capable of.
```

Author: done

+
To better understand the utilization of object storage by Loki, you can use the following query in the *Metrics* dashboard in the {ocp-product-title} web console:
+
[source]
----
sum by(status, container, operation) (label_replace(rate(loki_s3_request_duration_seconds_count{namespace="openshift-logging"}[5m]), "status", "${1}xx", "status_code", "([0-9]).."))
----

> * Ensure that automatic stream sharding is enabled for Loki. For information see "Enabling automatic stream sharding".

Reviewer: It would be nice if we could add the link to "Enabling automatic stream sharding"

> * If you use the OpenTelemetry Protocol (OTLP) data model, ensure that you configure the `streamLabels` resource in LokiStack. For more information, see “The importance of labels”.

Reviewer: ```suggestion
* If you use the OpenTelemetry Protocol (OTLP) data model. Remember that you can configure extra `streamLabels` in LokiStack to improve query performance. For more information, see “Best practices for Loki labels”.
```

> * If you use the OpenTelemetry Protocol (OTLP) data model, you can configure additional stream labels in LokiStack. For more information, see “Best practices for Loki labels”.

Reviewer 2: Hello @theashiot ,
It would be nice if we could add the link to “Best practices for Loki labels”.

Author: Thanks, @prithvipatil97, added
